src/main/java/uk/gov/hmcts/reform/authorisation/filters/ServiceAuthFilter.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.authorisation.filters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.filter.OncePerRequestFilter;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.exceptions.ServiceException;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ServiceAuthFilter extends OncePerRequestFilter {
+
+    public static final String AUTHORISATION = "ServiceAuthorization";
+    private static final Logger LOG = LoggerFactory.getLogger(ServiceAuthFilter.class);
+    private final List<String> authorisedServices;
+
+    private final AuthTokenValidator authTokenValidator;
+
+    @Autowired
```

Comment:
you might need @component annotation for auto-wire or leave filter bean definition to calling services including @Value.

Reply:
Good point, I have left he ServiceFilter as is and instead created the bean using AutoConfiguration that enables with conditional property.